amethyst_rendy/src/lib.rs

```diff
@@ -35,10 +35,14 @@
 //! * [`JointTransforms`](skinning::JointTransforms)
 //! * [`SpriteRender`](sprite::SpriteRender)
 
-#![allow(dead_code)]
-#![allow(unused_variables)]
-#![warn(missing_docs)]
-#![warn(missing_debug_implementations)]
+#![warn(
+    missing_debug_implementations,
+    missing_docs,
+    rust_2018_idioms,
+    rust_2018_compatibility
+)]
+#![warn(clippy::all)]
+#![allow(unused_variables, dead_code, clippy::new_without_default)]
```

Comment:
Why should `new_without_default` be allowed?

Reply:
Because we have multiple types which do this and this is not a pass to make significant changes everywhere to pass lints. I've made simple corrections where possible, and otherwise allowed lints where not. Because we now have flags for allowances, they can be fixed at a future time. 

If future work on the lints is not acceptable, then it's best to remove all this and close the PR.

Reply:
Why not just leave the warnings instead of hiding them? Seems like it would just makes it less likely for someone to notice them and fix them properly.

Reply:
To allow for only warnings for development but requiring the warnings to fail CI means we have to keep the warnings. But to pass CI, the warnings then must be dealt with some way since we can't distinguish between them. And we can't use deny, because them it makes development annoying

Reply:
Maybe allow the lints that currently don't pass in CI with `cargo clippy -- -A clippy::some_lint ... -D warnings`? That way they are still visible but don't stop CI from passing.